components/channel_select/channel_select.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+
+import Constants from 'utils/constants';
+import * as Utils from 'utils/utils.jsx';
+
+type Props = {
+    channels: Array<{
+        id: string;
+        display_name?: string;
+        name: string;
+        type: string;
+    }>;
+    onChange: () => void;
+    value: string;
+    selectOpen: boolean;
+    selectPrivate: boolean;
+    selectDm: boolean;
+};
+
+type OptionsState = Array<{
+    key: string;
+    value: string;
+    text: string;
+}>;
+
+function ChannelSelect({
+    channels,
+    onChange,
+    value,
+    selectOpen = false,
+    selectPrivate = false,
+    selectDm = false,
+}: Props): JSX.Element {
+    const [options, setOptions] = useState<OptionsState>([
+        {
+            key: '',
+            value: '',
+            text: Utils.localizeMessage('channel_select.placeholder', '--- Select a channel ---'),
+        },
+    ]);
+
+    useEffect(() => {
```

Comment:
I'd also prefer if we store <Options> component into the array rather than it's values. In our return function,
we can then do a {options} (as it's done in the original file) rather than having to do a map. In my opinion, I believe it makes for more readable code that way.